tests/Page/CKANRegistryPageControllerTest.php

```diff
@@ -50,17 +51,16 @@ public function testCKANClientConfigDecodesResultConditions()
         $config = $controller->getCKANClientConfig($page);
 
         $this->assertNotEmpty($config['fields']);
-        $this->assertContains([
-            'OriginalLabel' => 'city',
-            'ReadableLabel' => 'City',
-            'ShowInResultsView' => 1,
-            'ShowInDetailView' => 1,
-            'DisplayConditions' => [
-                ['match-select' => '1', 'match-text' => 'Auckland']
-            ],
-            'RemoveDuplicates' => 0,
-            'Type' => 'text',
-        ], $config['fields'], 'DisplayConditions are decoded from stored JSON format');
+        $f = $config['fields'][0];
+        $this->assertSame('city', $f['OriginalLabel']);
+        $this->assertSame('City', $f['ReadableLabel']);
+        $this->assertSame(true, $f['ShowInResultsView']);
+        $this->assertSame(true, $f['ShowInDetailView']);
+        $this->assertSame(false, $f['RemoveDuplicates']);
```

Comment:
The way it is is fine, but this might have been a more specific option.

```suggestion
        $this->assertTrue($f['ShowInResultsView']);
        $this->assertTrue($f['ShowInDetailView']);
        $this->assertFalse($f['RemoveDuplicates']);
```

Reply:
Technically, we're now testing something slightly different.

`assertSame` does a strict comparison.  I guess `assertContains` was doing a not strict comparison, because `1` and `true` would give you the same results.

Because we're just testing the first entry in `DisplayConditions` now, I guess the test would not fail if we got two entries all the sudden.

I'm OK with the test as re-written. Just observing some slight nuances.